lib/resque_bus/tasks.rb

```diff
@@ -1,106 +1,128 @@
 # require 'resque_bus/tasks'
 # will give you these tasks
 
-require "queue_bus/tasks"
-require "resque/tasks"
+require 'queue_bus/tasks'
+require 'resque/tasks'
 
 namespace :resquebus do
-  # deprecated
-  task :setup => ["queuebus:setup"] do
-    ResqueBus.note_deprecation "[DEPRECATION] rake resquebus:setup is deprecated. Use rake queuebus:setup instead."
+
+  task setup: ['queuebus:setup'] do
+    ResqueBus.note_deprecation '[DEPRECATION] rake resquebus:setup is deprecated. Use rake queuebus:setup instead.'
   end
 
-  task :driver => ["queuebus:driver"] do
-    ResqueBus.note_deprecation "[DEPRECATION] rake resquebus:driver is deprecated. Use rake queuebus:driver instead."
+  task driver: ['queuebus:driver'] do
+    ResqueBus.note_deprecation '[DEPRECATION] rake resquebus:driver is deprecated. Use rake queuebus:driver instead.'
   end
 
-  task :subscribe => ["queuebus:subscribe"] do
-    ResqueBus.note_deprecation "[DEPRECATION] rake resquebus:subscribe is deprecated. Use rake queuebus:subscribe instead."
+  task subscribe: ['queuebus:subscribe'] do
+    ResqueBus.note_deprecation '[DEPRECATION] rake resquebus:subscribe is deprecated. Use rake queuebus:subscribe instead.'
   end
 
-  task :unsubsribe => ["queuebus:unsubsribe"] do
-    ResqueBus.note_deprecation "[DEPRECATION] rake resquebus:driver is deprecated. Use rake queuebus:unsubsribe instead."
+  task unsubsribe: ['queuebus:unsubsribe'] do
+    ResqueBus.note_deprecation '[DEPRECATION] rake resquebus:driver is deprecated. Use rake queuebus:unsubsribe instead.'
   end
 end
 
+namespace :resque do
+  Rake::Task[:work].enhance ['queuebus:set_redis_server']
+end
+
 namespace :queuebus do
 
-  desc "Setup will configure a resque task to run before resque:work"
-  task :setup => [ :preload ] do
+  desc 'Sets the resque worker to use the QueueBus Redis server'
+  task :set_redis_server do
+    unless ENV['USE_QUEUEBUS_REDIS'].nil?
```

Comment:
I need to explain how the system works for this to make sense.
- `QUEUEBUS_REDIS_URL` needs to be set in the environment of any system that talks to QueueBus.
- Any such system in existence today also has a Resque instance (different than the QueueBus instance).  This means they will also have a Resque instance (currently assigned to `OPENREDIS_URL`)
- All QueueBus workers are just Resque workers.  You fire off the Resque worker task and that's it.
- If that's the case, how does a Resque worker know what queues to work?
- The environment variable `QUEUES` dictates this.  All the `queuebus:setup`, `queuebus:driver` tasks do is set an environment variable so that the Resque worker knows what queues to work.
- By introducing different servers, we are extending this pattern of passing information from one rake task to another.  In this case, the environment variable we're setting is whether or not to use the QueueBus redis instance, or the default.
- We know when we're hitting a QueueBus rake target that we should be using the QueueBus instance, so we record that in the environment, (**only for this dyno!**), and enhance the existing resque:work task to look for it and switch Redis instances when it sees it.

Does this help?